### PR TITLE
refactoring to react-popper 2.0

### PR DESCRIPTION
**Pull request by fliptaboada**
The `Dropdown` still doesn't work properly, it always appears above the anchor.

I went this route because `react-popper` now has its own `usePopper` hook, and I thought it would be better to use theirs.

But I found a little weird to work with `Ref` using `useState`. 
This is needed because `useRef` doesn't trigger rerenders and `useState` does, but it becomes incompatible with existing apis that use `RefObject` and this may be a problem.

What do you think @CarlosBonetti?
﻿

### CODE REVIEW

**Comment by CarlosBonetti** [comment]
It's a little weird but looks like the way to go. Regarding the RefObject possible problems, maybe we can change to React.Ref that supports both RefObject and a callback function.

File: src/components/Dropdown/Dropdown.test.tsx [hunk 1]
```diff
@@ -20,77 +19,106 @@ const DropdownTest = (props: Partial<DropdownProps> = {}) => {
...
 
 it('should call "onClose" when focus go outside menu', async () => {
   const close = jest.fn()
-  render(<DropdownTest open={true} onClose={close} />)
+  await act(async () => {
+    render(<DropdownTest open={true} onClose={close} />)
+  })
 
-  expect(document.activeElement).toEqual(document.body)
+  // expect(document.activeElement).toEqual(document.body)
```
Comment (fliptaboada) [inline src/components/Dropdown/Dropdown.test.tsx:107]: this is throwing an odd error, may be related to https://github.com/facebook/react/issues/18758

File: src/components/Dropdown/Dropdown.test.tsx [hunk 2]
```diff
@@ -20,77 +19,106 @@ const DropdownTest = (props: Partial<DropdownProps> = {}) => {
...
 
   expect(close).not.toHaveBeenCalled()
 
   fireEvent.blur(document.body.querySelector('ul'))
-  await wait()
-  expect(close).toHaveBeenCalledTimes(1)
+  await waitFor(() => expect(close).toHaveBeenCalledTimes(1))
```
Comment (fliptaboada) [inline src/components/Dropdown/Dropdown.test.tsx:112]: This assertion is still failing, not sure why, `blur` fires correctly but `activeElement` stays an `li` from the `DropdownMenu`.

It works correctly on the storybook, tho.

File: src/components/Dropdown/Dropdown.test.tsx (hunk 1 above)
Comment (fliptaboada) [reply src/components/Dropdown/Dropdown.test.tsx]: the assertion was incorrect, and that was causing the error. 
the message just wasn't clear.

File: src/components/Dropdown/Dropdown.test.tsx (hunk 2 above)
Comment (fliptaboada) [reply src/components/Dropdown/Dropdown.test.tsx]: changed to `element.focus()` to replicate the focus/blur correctly.

**Comment by calistro** [comment]
@CarlosBonetti It's all looking good to me, but will be safer if you could take a look too, do you mind?

**Comment by CarlosBonetti** [comment]
@calistro Looks good to me. Make sure to test the scenario of a popper inside a modal. A DatePicker inside a modal, for example. We had this error before where the modal would create an internal scroll to fit the popper size.

**Comment by calistro** [comment]
:rocket:

**Pull request merged**